src/wallet/rpcwallet.cpp

```diff
@@ -2553,6 +2561,19 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
             feeRate = CFeeRate(AmountFromValue(options["feeRate"]));
             overrideEstimatedFeerate = true;
         }
+
+        if (options.exists("subtractFeeFromAmount")) {
+            subtractFeeFromAmount = options["subtractFeeFromAmount"].get_array();
+            for (unsigned int idx = 0; idx < subtractFeeFromAmount.size(); idx++) {
+                string strAddress(subtractFeeFromAmount[idx].get_str());
+                CBitcoinAddress address(strAddress);
+                if (!address.IsValid())
+                    throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, string("Invalid Bitcoin address: ")+strAddress);
+                if (setSubtractFeeFromAmount.count(strAddress))
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ")+strAddress);
+                setSubtractFeeFromAmount.insert(strAddress);
```

Comment:
At the time of running `fundrawtransaction` there is no change output, and the user wouldn't know where the change will be inserted, so the position marking must be done before the change output is added.

I think since it is possible to add arbitrary hex output scripts which may not even have a corresponding address we need to be able to address the outputs by number rather than by address. It's also kind of ugly having to give the same address twice, once to createrawtransaction and then again to fundrawtransaction. I think using the output index (0 based) is cleaner.